src/SharpLearning.Backend.Cntk.Test/MnistCnnExampleTest.cs

```diff
@@ -92,12 +96,154 @@ public void MnistCnnTest()
                     }
 
                     var samplesSeenNextBatch = (i + 2) * minibatchSize;
-                    if (samplesSeenNextBatch >= epochSize)
+                    if (samplesSeenNextBatch >= numberOftrainingSamples)
                     {
                         readerTrain.Reset();
                     }
                 }
             }
+
+            // Test model.
+            var csharpError = MnistSimpleExampleTest.TestModelMnistLoader(z, device, mnist);
+            Trace.WriteLine($"Test Error: {csharpError}");
+
+            // Save model.
+            var modelPath = "cnn_mnist_csharp_loader.dnn";
+            z.Save(modelPath);
+
+            // Test loaded model.
+            var loadedModel = Function.Load(modelPath, device);
+            var loadedModelError = MnistSimpleExampleTest.TestModelMnistLoader(loadedModel, device, mnist);
+
+            Trace.WriteLine("Loaded Model Error: " + loadedModelError);
+            Assert.AreEqual(csharpError, loadedModelError, 0.00001);
+
+            // Test against python example.
+            var pythonError = 0.89;
+            Assert.AreEqual(pythonError, csharpError, 0.00001);
+        }
+
+        const string FeatureId = "features";
+        const string LabelsId = "labels";
+
+        [TestMethod]
+        public void MnistCnnTest_MinibatchSource()
+        {
+            // Set global data and device type. 
+            var dataType = CntkDataType.Float;
+            DeviceDescriptor device = DeviceDescriptor.CPUDevice;
+            Trace.WriteLine("Using device: " + device.Type + " with data type: " + dataType);
+
+            // Define data.
+            var dataDirectoryPath = @"..\..\..\python\src\CntkPython\";
+            var imageHeight = 28;
+            var imageWidth = 28;
+            var numChannels = 1;
+            var numOutputClasses = 10;
+            var inputDimensions = new int[] { imageWidth, imageHeight, numChannels };
+
+            // Input variables denoting the features and label data.
+            Variable inputVar = Variable.InputVariable(inputDimensions, dataType); ;
+            Variable labelVar = Variable.InputVariable(new int[] { numOutputClasses }, dataType);
+
+            // Instantiate the feedforward classification model
+            var scaledInput = CNTKLib.ElementTimes(Constant.Scalar(dataType, 0.00390625, device), inputVar);
+
+            // setup initializer
+            var init = CNTKLib.UniformInitializer(
```

Comment:
@nietras I was **not** able to get deterministic results using the recommended `GlorotUniformInitializer`. So I had to use the `UniformInitializer` instead. This results in higher error rate but deterministic behavior, which should be the most important for testing.

Reply:
@mdabros we need to change initializers to be the same for both TF and CNTK anyway so, and I still have not found out why C# and python TF is different for the deep case, but it seems to be related to random variable initialization, although they use the same seeds now.